Chargebee/Classes/Cache/CBCache.swift

```diff
@@ -0,0 +1,91 @@
+//
+//  CBCache.swift
+//  Chargebee
+//
+//Created by ramesh_g on 28/07/23.
+//
+
+import Foundation
+
+
+private struct ConfigCacheModel: Codable {
+    var config: CBAuthentication
+    var validTill: Date
+    init(config: CBAuthentication, validTill: Date) {
+        self.config = config
+        self.validTill = validTill
+    }
+}
+
+private protocol CacheProtocol {
+    func writeConfigDetails(object:CBAuthentication)
+    func readConfigDetails(logger: CBLogger, handler: @escaping CBAuthenticationHandler)
+    func saveAuthenticationDetails(data: CBAuthenticationStatus)
+    func isCacheDataAvailable()-> Bool
+}
+
+internal struct CBCache: CacheProtocol {
+    private let uniqueIdKey = "chargebee_config"
+    static let shared = CBCache()
+    
+    internal func writeConfigDetails(object configObject:CBAuthentication) {
+        let configObj: ConfigCacheModel = ConfigCacheModel(config: configObject, validTill: createTime())
+        if let encoded = try? JSONEncoder().encode(configObj) {
+            UserDefaults.standard.set(encoded, forKey: getFormattedkey())
+        }
+    }
+    
+    private func createTime() -> Date {
+        let currentDate = Date()
+        let newDate = NSDate(timeInterval: 86400, since: currentDate)
+        return newDate as Date
+    }
+    
+    internal func readConfigDetails(logger: CBLogger, handler: @escaping CBAuthenticationHandler){
+        let (onSuccess, _) = CBResult.buildResultHandlers(handler, logger)
+        if let data = UserDefaults.standard.object(forKey: getFormattedkey()) as? Data,
+           let cacheModel = try? JSONDecoder().decode(ConfigCacheModel.self, from: data) {
+            let cacheObj = ConfigCacheModel(config: cacheModel.config, validTill: cacheModel.validTill)
+            let auth = CBAuthenticationStatus.init(details: cacheObj.config)
+            onSuccess(auth)
+        }
```

Comment:
Can we handle for Error case as well? 
Lets clear the cache and let the SDK call the auth APi in case of errors

Reply:
> Can we handle for Error case as well? Lets clear the cache and let the SDK call the auth APi in case of errors

Error handling done updated